SimulationAgent/Agent.cs

```diff
@@ -250,18 +254,10 @@ private async Task SaveSimulationStatisticsAsync(IList<Simulation> simulations)
             {
                 foreach (var simulation in simulations)
                 {
-                    try
+                    if (this.simulationManagers.ContainsKey(simulation.Id))
```

Comment:
can simulationManagers be null? 

I think we ran into an issue, with a new deployment, when you create first simulation this was failing. @avanikp - I think you worked on that fix, can you provide more details?

Reply:
iirc that, or something similar, was a bug I reported to @avanikp during a CR and was fixed. 

The list of managers is a readonly collection initialized in the ctor so it won't be null